### DISCUSSION TITLE
Will the German system be expanded?

### DISCUSSION THREAD

**Discussion by HausBJB** [DevTrevi/DgcReader#70] [Q&A] [answered]
Hello, I have a question. Will the German system be expanded? E.g. the blacklist or 2G+ function. Or because it is not supported, are we not doing anything anymore? thank you

**Comment by DevTrevi** [comment, marked as answer]
Hi @HausBJB, it will be great to implement the latest functionality for the German validation provider, but unfortunately I have very little time available to work on these features at the moment, especially to get all the required informations about how these features are implemented in the German App and how they apply to the other 18 countries supported by the app...

I have briefly checked the latest updates on the German app and If I correctly understand, it always validate the greenpass with the 3G rules, then applying some hardcoded checks to exclude non-2G entries (like tests entries), but I'm not sure if these checks are the same for all the supported countries or if there are some exceptions...
If someone living in Germany could help by giving me at least the specifications, I could take in consideration to implement these missing features, but I'm a little constrained by available time

**Comment by HausBJB** [comment]
Ok first of all thanks for your answer. I understand you don't have time. That was just a question of the future. I'm trying to get some info out. I think the blacklist is also hard-coded and there are currently only 2 pharmacies that are blocked
